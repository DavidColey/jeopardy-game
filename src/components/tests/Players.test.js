import React from "react";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-15";
import { configure, shallow } from "enzyme";

import { Players } from "./../Players";

describe("Player Component", () => {
  beforeAll(() => {
    configure({ adapter: new Adapter() });
  });

  it("renders correctly when not passing players", () => {
    const tree = renderer.create(<Players store={{}} players={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when passing players in", () => {
    const tree = renderer
      .create(<Players players={[{ name: "John" }, { name: "William" }]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Calls addPlayer when enter is pressed", () => {
    const mockAddPlayer = jest.fn();
    const wrapper = shallow(
      <Players
        addPlayer={mockAddPlayer}
        players={[{ name: "John" }, { name: "William" }]}
      />
    );
    wrapper.instance().handleInput({ keyCode: 13, target: { value: "John" } });
    expect(mockAddPlayer.mock.calls.length).toBe(1);
  });

  it("Doesn't call addPlayer when enter is not pressed", () => {
    const mockAddPlayer = jest.fn();
    const wrapper = shallow(
      <Players
        addPlayer={mockAddPlayer}
        players={[{ name: "John" }, { name: "William" }]}
      />
    );
    wrapper.instance().handleInput({ keyCode: 12, target: { value: "John" } });
    expect(mockAddPlayer.mock.calls.length).toBe(0);
  });
});
