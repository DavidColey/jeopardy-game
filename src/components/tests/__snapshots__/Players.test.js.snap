// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Component renders correctly when not passing players 1`] = `
<div>
  <h2>
    Players
  </h2>
  <input
    onKeyUp={[Function]}
    placeholder="Add player..."
    type="text"
  />
</div>
`;

exports[`Player Component renders correctly when passing players in 1`] = `
<div>
  <h2>
    Players
  </h2>
  <p
    className="player-text"
  >
    John
  </p>
  <p
    className="player-text"
  >
    William
  </p>
  <input
    onKeyUp={[Function]}
    placeholder="Add player..."
    type="text"
  />
</div>
`;
